contracts/integration/LockboxAdapter.sol

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {IXERC20} from "../shared/IXERC20/IXERC20.sol";
+import {IXERC20Lockbox} from "../shared/IXERC20/IXERC20Lockbox.sol";
+import {IConnext} from "@connext/interfaces/core/IConnext.sol";
+import {IXReceiver} from "@connext/interfaces/core/IXReceiver.sol";
+
+interface IXERC20Registry {
+  function getXERC20(address erc20) external view returns (address xerc20);
+
+  function getERC20(address xerc20) external view returns (address erc20);
+
+  function getLockbox(address erc20) external view returns (address xerc20);
+}
+
+contract LockboxAdapter is IXReceiver {
+  address immutable connext;
+  address immutable registry;
```

Comment:
`connext` and `registry` variables should be `internal`?
```suggestion
  address public immutable registry;
```

Reply:
No not necessarily.